modules/installation-special-config-encrypt-disk.adoc

```diff
@@ -0,0 +1,89 @@
+// Module included in the following assemblies:
+//
+// * installing/install_config/installing-customizing.adoc
+
+[id="installation-encrypting-disks_{context}"]
+= Encrypting disks during installation
+
+During {product-title} installation, you can enable disk encryption on all master and worker nodes.
+This feature:
+
+* Is available for installer provisioned infrastructure
+and user provisioned infrastructure deployments
+* Is supported on Red Hat Enterprise Linux CoreOS (RHCOS) systems only
+* Sets up disk encryption during the manifest installation phase so all data written to disk, from first boot forward, is encrypted
+* Encrypts data on the root filesystem only (/dev/mapper/coreos-luks-root on /)
+* Requires no user intervention for providing passphrases
+* Uses AES-256-CBC encryption
+* Should be enabled for your cluster to be FIPS compliant.
+
+There are two different supported encryption modes:
+
+* TPM v2: This is the preferred mode. TPM v2 stores passphrases in a secure cryptoprocessor.
+To implement TPM v2 disk encryption, create an Ignition config file as described below.
+
+* Tang: To use link:https://github.com/latchset/tang[Tang] to encrypt your cluster, you need to use a Tang server.
+link:https://github.com/latchset/clevis[Clevis] implements decryption on the client side.
+Tang encryption mode is only supported for bare metal installs.
+
+Follow one of the two procedures below to enable disk encryption for the nodes in your cluster.
+
+== Enabling TPM v2 disk encryption
+Use this procedure to enable TPM v2 mode disk encryption during {product-title} deployment.
+
+.Procedure
+
+. Check to see if TPM v2 encryption needs to be enabled in the BIOS on each node.
+This is required on most Dell systems.
+
+. Generate the Kubernetes manifests for the cluster:
++
+----
+$ ./openshift-install create manifests --dir=<installation_directory>
+----
+
+. In the `openshift` directory, create a file (for example, 99_openshift-tpmv2-encryption.yaml)
+to define the `/etc/clevis.json` file. This is how that file
+can be configured:
++
+----
+ "storage": {
```

Comment:
The file is `99_openshift-tpmv2-encryption.yaml`  but the contents here is JSON.  Are you sure this is the correct contents?

Reply:
After looking at it again, my suggestion isn't correct because the `master.ign `, `worker.ign`, and `bootstrap.ign` grab the file from the api server.

Reply:
OK, after some testing, I got it to work.   I created the following MachineConfigs in the `openshift` directory

99_openshift-master-tpmv2-encryption.yaml:
```
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  name: master-tpm
  labels:
    machineconfiguration.openshift.io/role: master
spec:
  config:
    ignition:
      version: 2.2.0
    storage:
      files:
      - contents:
          source: data:text/plain;base64,e30K
        filesystem: root
        mode: 420
        path: /etc/clevis.json
```
99_openshift-worker-tpmv2-encryption.yaml:
```
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  name: worker-tpm
  labels:
    machineconfiguration.openshift.io/role: worker
spec:
  config:
    ignition:
      version: 2.2.0
    storage:
      files:
      - contents:
          source: data:text/plain;base64,e30K
        filesystem: root
        mode: 420
        path: /etc/clevis.json
```

Reply:
I will use these two example.

Reply:
I'm going to drop the .ign example and just show your MachineConfig examples.

Reply:
I used the same basic format you used on the tpm example in the tang example as well, but added the longer source.